ci/run-validator.sh

```diff
@@ -34,35 +46,59 @@ wait_for_block() {
 
 # Saves the hash of the genesis node and the keys funded with the coins
 # to the directory shared with the bridge node
-provision_bridge_node() {
+provision_bridge_nodes() {
   local genesis_hash
-  local user_address
+  local last_node_idx=$((BRIDGE_COUNT - 1))
 
   # Save the genesis hash for the bridge
   genesis_hash=$(wait_for_block 1)
   echo "Saving a genesis hash to $GENESIS_HASH_FILE"
   echo "$genesis_hash" > "$GENESIS_HASH_FILE"
 
-  # Create a new user account
-  echo "Creating a new keys for the test user"
-  celestia-appd keys add "$USER_NAME" --keyring-backend "test"
-  user_address="$(celestia-appd keys show "$USER_NAME" -a --keyring-backend="test")"
+  # Get or create the keys for bridge nodes
+  for node_idx in $(seq 0 "$last_node_idx"); do
+    local bridge_name="bridge-$node_idx"
+    local key_file="$CREDENTIALS_DIR/$bridge_name.key"
+    local addr_file="$CREDENTIALS_DIR/$bridge_name.addr"
+
+    if [ ! -e "$key_file" ]; then
+      # if key don't exist yet, then create and export it
+      echo "Creating a new keys for the $bridge_name"
+      celestia-appd keys add "$bridge_name" --keyring-backend "test"
+      # export it
+      echo "password" | celestia-appd keys export "$bridge_name" 2> "$key_file"
+      # export associated address
+      node_address "$bridge_name" > "$addr_file"
+    else
+      # otherwise, just import it
+      echo "password" | celestia-appd keys import "$bridge_name" "$key_file" \
+        --keyring-backend="test"
+    fi
+  done
 
-  # Send it the coins.
-  # Coins transfer need to be after validator registers EVM address, which starts in block 2.
-  # We actually give it a bit more time to make sure it has finished.
+  # Transfer the coins to bridge nodes addresses
+  # Coins transfer need to be after validator registers EVM address, which happens in block 2.
   # see `setup_private_validator`
-  wait_for_block 5
-  echo "Transfering coins to the test user"
-  echo "y" | celestia-appd tx bank send \
-    "$VALIDATOR_NAME" \
-    "$user_address" \
-    "$USER_COINS" \
-    --fees 21000utia
-
-  # And export it for the bridge
-  echo "Exporting the keys for a test user to $USER_KEY_FILE"
-  echo "password" | celestia-appd keys export "$USER_NAME" 2> "$USER_KEY_FILE"
+  local start_block=3
+
+  for node_idx in $(seq 0 "$last_node_idx"); do
+    # TODO: create an issue in celestia-app and link it here
+    # we need to transfer the coins for each node in separate
+    # block, or the signing of all but the first one will fail
+    wait_for_block $((start_block + node_idx))
+
+    local bridge_name="bridge-$node_idx"
+    local bridge_address
+
+    bridge_address=$(node_address "$bridge_name")
```

Comment:
```suggestion
    local bridge_address=$(node_address "$bridge_name")
```
or was there a reason you've put them separately?

Reply:
yeah, so if you do a local binding and command substitiution at once, the error from command substitution will be suppressed by the exit code of local binding

Reply:
til! thanks